kyaml/kio/filters/merge3.go

```diff
@@ -143,27 +172,36 @@ func (ts *tuples) isSameResource(meta1, meta2 yaml.ResourceMeta) bool {
 
 // add adds a node to the list, combining it with an existing matching Resource if found
 func (ts *tuples) add(node *yaml.RNode) error {
-	nodeMeta, err := node.GetMeta()
-	if err != nil {
-		return err
-	}
 	for i := range ts.list {
 		t := ts.list[i]
-		if ts.isSameResource(t.meta, nodeMeta) {
+		if ts.matcher == nil {
+			ts.matcher = &DefaultGVKNNMatcher{MergeOnPath: true}
+		}
+		if ts.matcher.IsSameResource(addedNode(t), node) {
 			return t.add(node)
 		}
 	}
-	t := &tuple{meta: nodeMeta}
+	t := &tuple{}
 	if err := t.add(node); err != nil {
 		return err
 	}
 	ts.list = append(ts.list, t)
 	return nil
 }
 
+// addedNode returns one on the existing added node in the tuple
```

Comment:
Maybe add a comment explaining the order in which we check the elements of the tuple?

Reply:
The order doesn't matter here. We just want one of the non nil nodes out of the 3.